Makefile

```diff
@@ -0,0 +1,157 @@
+	@echo "\tCUSTOM_MOZCONFIG: extends supplied mozconfig with user parameters (e.g: |mk_add_options MOZ_MAKE_FLAGS="-j3 -s"|)"
+	@echo ""
+
+${BUILD_DIR}:
+	mkdir -p ${BUILD_DIR}
+
+##### Install build dependencies
+${BUILD_DIR}/.build_deps_ready: ${BUILD_DIR}
+	python ${mkfile_dir}/bootstrap.py --application-choice desktop
+	sudo apt-get install libgstreamer-plugins-bad1.0-dev \
+		libgstreamer-plugins-base0.10-dev libgstreamer-plugins-base1.0-dev \
+		libgstreamer1.0-dev
+	touch ${BUILD_DIR}/.build_deps_ready
+
+##### Install runtime dependencies
+${BUILD_DIR}/.runtime_deps_ready:
+	sudo apt-get install xserver-xephyr
+	touch ${BUILD_DIR}/.runtime_deps_ready
+
+##### Clone gecko source if missing
+${GECKO_SOURCE_DIR}:
+	git clone -b ronin \
+		https://github.com/Phoxygen/gecko-dev ${GECKO_SOURCE_DIR}
+
+# (Todo: check that Ronin root commit is an ancestor of the active branch)
+gecko-sources: ${GECKO_SOURCE_DIR}
+
+##### Clone gaia source if missing
+${GAIA_SOURCE_DIR}:
+	git clone -b ronin \
+		https://github.com/Phoxygen/gaia ${GAIA_SOURCE_DIR}
+
+# Todo: check that Ronin root commit is an ancestor of the active branch)
+gaia-sources: ${GAIA_SOURCE_DIR}
+
+sources: gecko-sources gaia-sources
+
+##### Build gaia profile
+# (The built profile lives in ${BUILD_DIR}/profile)
+build-gaia:
+	if [ ! -d ${BUILD_DIR}/gaia/profile ]; then mkdir -p ${BUILD_DIR}/gaia/profile; fi
+	cd ${GAIA_SOURCE_DIR} && \
+	DEVICE_DEBUG=1 GAIA_DEVICE_TYPE=phone DESKTOP_SHIMS=1 NOFTU=1 PROFILE_DIR=${BUILD_DIR}/gaia/profile make
+
+${BUILD_DIR}/gaia/profile:
+	$(MAKE) build-gaia
+
+##### Prepare gecko build
+# Copy mozconfig to source dir, and append custom_mozconfig if existing
+# Note: 
+${GECKO_SOURCE_DIR}/mozconfig: ${mkfile_dir}/mozconfig ${CUSTOM_MOZCONFIG}
+	cp -v ${mkfile_dir}/mozconfig ${GECKO_SOURCE_DIR}/mozconfig
+	@if [ -n "${CUSTOM_MOZCONFIG}" ]; then \
+		cat ${CUSTOM_MOZCONFIG} >> ${GECKO_SOURCE_DIR}/mozconfig; \
+	fi
+
+###### Build gecko
+build-gecko: ${BUILD_DIR}/.build_deps_ready ${GECKO_SOURCE_DIR}/mozconfig
+	mkdir -p ${BUILD_DIR}/gecko && \
+	cd ${GECKO_SOURCE_DIR} && \
+	MOZ_OBJDIR=${BUILD_DIR}/gecko ./mach build && \
+	MOZ_OBJDIR=${BUILD_DIR}/gecko ./mach package
+
+${BUILD_DIR}/gecko/dist/b2g:
+	$(MAKE) build-gecko
+
+###### Build meta-goal
+build: ${BUILD_DIR}/.build_deps_ready sources build-gaia build-gecko
+
+${B2G_PROFILE_PATH}:
+	mkdir -p ${B2G_PROFILE_PATH}
+
+###### Run b2g
+# Uses Xephyr to ease testing.
+# It doens't depend on the build goal, instead it only relies on
+# build outputs (b2g dist folder and gaia profile). Otherwise 
+# running |make run| takes too much time.
+# b2g depends on ~/.mozilla/b2g/xsession.profile so we need to copy
+# gaia profile there
+#run: ${BUILD_DIR}/.runtime_deps_ready ${BUILD_DIR}/gecko/dist/b2g ${B2G_PROFILE_PATH} ${BUILD_DIR}/gaia/profile
+run: ${B2G_PROFILE_PATH}
+	cp -a ${BUILD_DIR}/gaia/profile/* ${B2G_PROFILE_PATH}/
```

Comment:
this omits every dot file. Is it what you want? I would have used:

```bash
cp -R ${BUILD_DIR}/gaia/profile ${B2G_PROFILE_PATH}
```
Unless there is already something in the destination folder?